erpnext/public/js/controllers/buying.js

```diff
@@ -213,6 +212,22 @@ erpnext.buying.BuyingController = erpnext.TransactionController.extend({
 		this.qty(doc, cdt, cdn);
 	},
 
+	rate: function(doc,cdt,cdn) {
+		this.calculate_stock_uom_rate(doc, cdt, cdn);
+		this._super(doc, cdt, cdn);
+	},
+
+	conversion_factor: function(doc,cdt,cdn) {
+		this.calculate_stock_uom_rate(doc, cdt, cdn);
+		this._super(doc, cdt, cdn);
+	},
+
+	calculate_stock_uom_rate: function(doc, cdt, cdn) {
+		let item = frappe.get_doc(cdt, cdn);
+		item.stock_uom_rate = item.rate/item.conversion_factor;
+		refresh_field("stock_uom_rate", item.name, item.parentfield);
```

Comment:
Should change on changing UOM too. Does not happen currently as seen in Purchase Order:
1200/15 = 80, not 5.33. It seems to show 80/15, which is not the right rate.
![Screenshot 2021-01-19 at 9 11 12 PM](https://user-images.githubusercontent.com/25857446/105056400-25c40980-5a9a-11eb-9c33-a236c1435b1d.png)